make.js

```diff
@@ -177,222 +177,59 @@ CLI.gendocs = function() {
-                    return `file:${path.relative(taskPath, commonPack.packFilePath)}`;
-                });
-                run(`npm install --save-exact ${installPaths.join(' ')}`);
-            }
-        }
-
-        // build Node task
-        if (shouldBuildNode) {
-            buildNodeTask(taskPath, outDir);
-        }
-
-        // remove the hashes for the common packages, they change every build
-        if (commonPacks.length) {
-            var lockFilePath = path.join(taskPath, 'package-lock.json');
-            if (!test('-f', lockFilePath)) {
-                lockFilePath = path.join(taskPath, 'npm-shrinkwrap.json');
-            }
-            var packageLock = fileToJson(lockFilePath);
-            Object.keys(packageLock.dependencies).forEach(function (dependencyName) {
-                commonPacks.forEach(function (commonPack) {
-                    if (dependencyName == commonPack.packageName) {
-                        delete packageLock.dependencies[dependencyName].integrity;
-                    }
-                });
-            });
-            fs.writeFileSync(lockFilePath, JSON.stringify(packageLock, null, '  '));
-        }
-
-        // copy default resources and any additional resources defined in the task's make.json
-        console.log();
-        console.log('> copying task resources');
-        copyTaskResources(taskMake, taskPath, outDir);
-
-        if (removeNodeModules) {
-            const taskNodeModulesPath = path.join(taskPath, 'node_modules');
-
-            if (fs.existsSync(taskNodeModulesPath)) {
-                console.log('\n> removing node modules');
-                rm('-Rf', taskNodeModulesPath);
-            }
-
-            const taskTestsNodeModulesPath = path.join(taskPath, 'Tests', 'node_modules');
+    const nodeVersion = argv.node ? argv.node : "Default";
+    const allTasksNode20 = allTasks.filter((taskName) => {
+        return taskName.endsWith("Node20");
+    });
+    const allTasksDefault = allTasks.filter((taskName) => {
+        return !taskName.endsWith("Node20");
+    });
 
-            if (fs.existsSync(taskTestsNodeModulesPath)) {
-                console.log('\n> removing task tests node modules');
-                rm('-Rf', taskTestsNodeModulesPath);
-            }
+    if (nodeVersion == "Node20") {
+        ensureTool('node', '--version', 'v20.3.1');
+        allTasksNode20.forEach(buildTasks);
+    } else if (nodeVersion == "Default") {
+        if (allTasksNode20) {
+            console.log(
+            "==========================================\n" + 
+            "IMPORTANT NOTE: There are additional tasks that need to be build with a different node configuration. \n" + 
+            "Unfortunately, we cannot switch node versions while running make.js.  Run the following commands to continue: \n" +
+            "nvm use 20.3.1\n" +
+            "node make.js build --task taskName --node Node20\n" + 
```

Comment:
>make.js build --task taskName --node Node20\ [](http://example.com/codeflow?start=18&length=44)

[not blocking, could do in another pr] it would be good to put in the exact command line to run, including the user's actual args so they can actually copy+paste